### DISCUSSION TITLE
Need help add a keybinding to insert "lastarg" (like ALT+.)

### DISCUSSION THREAD

**Discussion by VikJES1969** [akinomyoga/ble.sh#469] [Q&A]
Can someone tell how I can add a keybinding identical to ALT+. that inserts the lastarg of the previous command. I'm new to ble.sh and I can't figure out how to do it.
I'm actually surprised this keybinding isn't in ble.sh by default already.
Thank you in advance

**Comment by akinomyoga** [comment]
Alt+. (<kbd>M-.</kbd>) is already in the keymaps of ble.sh. It's working in my environment. It inserts the last word for the first call, and it cycles through words for successive calls.

- What happens in your environment when you press Alt+.?

Maybe it is related to the terminal encoding of keys.

- What is the result of the following commands in your environment?

```bash
$ ble/widget/display-shell-version
$ locale
$ builtin bind -v | grep meta
$ ble-bind -P | grep -F M-.
```

- I'd like to check what data is actually sent by your terminal when you press "Alt+.". Could you provide the result of the following command?

```bash
$ ble/debug/keylog#start
$ echo                          # <-- Please press Alt+. here.
$ ble/debug/keylog#end
```

**Comment by VikJES1969** [comment]
Hello,

Thanks for the super quick reply.

When I press Alt+. I get a "unbound keyseq: M-." message in the upper left corner of the terminal window.

Below is the output of the commands you've asked me to enter:

$ ble/widget/display-shell-version 
GNU bash, version 5.2.26(1)-release (x86_64-pc-linux-gnu) with ble.sh

$ locale 
LANG=en_CA.UTF-8
LC_CTYPE="en_CA.UTF-8"
LC_NUMERIC="en_CA.UTF-8"
LC_TIME="en_CA.UTF-8"
LC_COLLATE="en_CA.UTF-8"
LC_MONETARY="en_CA.UTF-8"
LC_MESSAGES="en_CA.UTF-8"
LC_PAPER="en_CA.UTF-8"
LC_NAME="en_CA.UTF-8"
LC_ADDRESS="en_CA.UTF-8"
LC_TELEPHONE="en_CA.UTF-8"
LC_MEASUREMENT="en_CA.UTF-8"
LC_IDENTIFICATION="en_CA.UTF-8"
LC_ALL=

$ builtin bind -v | grep meta
set convert-meta off
set enable-meta-key on
set input-meta on
set meta-flag on
set output-meta on

P.S. I was under the impression that since ble.sh doesn't use Readline, it wouldn't source the /etc/inputrc or .inputrc which to me looks like it did there, if you'd care to enlighten me, I'd appreciate it.

$ ble-bind -P | grep -F M-.
[ble: exit 1]

I didn't get very far with the last section of your email:
When I typed "$ ble/debug/keylog#start"
I got the following error:
λ ble/debug/keylog#start
bash: ble/debug/keylog#start: No such file or directory
[ble: exit 127]

I wonder if all of this is caused by the installation of the ble.sh package from the package found in the AUR.

Just to confirm it wasn't something strange with my setup, I built a very basic VM also running Arch and was able to replicate the problem (but I did install ble.sh using the AUR package).

In the meantime, I'll uninstall this AUR package and go through the installation from source using git (as detailed in the official GitHub page), I'll let you know if this resolves the problem. I wouldn't be surprised there some issue with the AUR package.

Thanks again.

**Reply by akinomyoga** [reply]
Ah, OK. You are using ble-0.3. Could you use ble-0.4? The corresponding Arch package is `blesh-git`.

**Comment by VikJES1969** [comment]
Yep, it worked. 
I used the blesh-git package instead and it's working like a charm.

Thank you so much for your help.
